gemfirexd/core/src/main/java/com/pivotal/gemfirexd/internal/impl/sql/GenericStatement.java

```diff
@@ -151,7 +151,7 @@ Licensed to the Apache Software Foundation (ASF) under one or more
             Pattern.compile(".*PUT\\s+INTO\\s+(TABLE)?.*SELECT\\s+.*",
                 Pattern.CASE_INSENSITIVE);
         private static final Pattern EXECUTION_ENGINE_STORE_HINT =
-            Pattern.compile(".*EXECUTIONENGINE(\\s+)?+=(\\s+)?+STORE\\s+.*",
+            Pattern.compile(".*EXECUTIONENGINE(\\s+)?+=(\\s+)?+STORE.*[\\r\\n].*",
```

Comment:
Is it necessary for the query to end in a newline? Both the old pattern as well as new one look problematic since there can be multiple hints separated by comma, and there might not be a terminating newline either. I think the better pattern is to check for word boundaries on both sides (and allowing for possible whitespaces): `.*\\bEXECUTIONENGINE(\\s+)?+=(\\s+)?+STORE\\s*\\b.*`